app/src/test/java/com/x8bit/bitwarden/MainViewModelTest.kt

```diff
@@ -306,6 +313,31 @@ class MainViewModelTest : BaseViewModelTest() {
         )
     }
 
+    @Suppress("MaxLineLength")
+    @Test
+    fun `on ReceiveFirstIntent with complete registration data should set the special circumstance to CompleteRegistration`() {
+        val viewModel = createViewModel()
+        val mockIntent = mockk<Intent>()
+        val completeRegistrationData = mockk<CompleteRegistrationData>()
+        every { mockIntent.getPasswordlessRequestDataIntentOrNull() } returns null
+        every { mockIntent.getAutofillSaveItemOrNull() } returns null
+        every { mockIntent.getCompleteRegistrationDataIntentOrNull() } returns completeRegistrationData
+        every { mockIntent.getAutofillSelectionDataOrNull() } returns null
+        every { intentManager.getShareDataFromIntent(mockIntent) } returns null
+        every { mockIntent.isMyVaultShortcut } returns false
+        every { mockIntent.isPasswordGeneratorShortcut } returns false
```

Comment:
⛏️ 🎨 Could define the mock Intent behavior within the initializer lambda. Cleans it up a little, imo. Don't feel obligated to change though. It works either way. 😄 

```
val mockCompleteRegistrationData = mockk<ComlpeteRegistrationData>()
val mockIntent = mockk<Intent> {
    every { getPasswordlessRequestDataIntentOrNull() } returns null
    every { getAutofillSaveItemOrNull() } returns null
    // etc...
}
```
or even make `mockIntent` "relaxed" with 
```
val mockIntent = mockk<Intent>(relaxed = true) {
    every { getCompleteRegistrationDataIntentOrNull() } returns mockData
}
```
which will make functions return default values (`null` and `false`) unless you set them to do otherwise.